libs/vkd3d/state.c

```diff
@@ -3687,7 +3714,7 @@ static HRESULT d3d12_pipeline_state_init_graphics(struct d3d12_pipeline_state *s
      * be exceedingly rare. */
     can_compile_pipeline_early =
             (desc->primitive_topology_type != D3D12_PRIMITIVE_TOPOLOGY_TYPE_PATCH || graphics->patch_vertex_count != 0) &&
-            desc->primitive_topology_type != D3D12_PRIMITIVE_TOPOLOGY_TYPE_UNDEFINED;
+            (desc->primitive_topology_type != D3D12_PRIMITIVE_TOPOLOGY_TYPE_UNDEFINED || (graphics->stage_flags & VK_SHADER_STAGE_MESH_BIT_EXT));
```

Comment:
Reading this again, there might be a problem if the PSO sets PRIMITIVE_TOPOLOGY_TYPE_PATCH here, since it should be ignored. Might be better to always allow early compile with MESH.